core/src/main/java/hudson/Functions.java

```diff
@@ -1933,18 +1932,21 @@ public static String toEmailSafeString(String projectName) {
      * @deprecated use {@link JNLPLauncher#getInboundAgentUrl}
      */
     @Deprecated
-    public String getServerName() {
+    public String getServerName() throws URISyntaxException, MalformedURLException {
         // Try to infer this from the configured root URL.
         // This makes it work correctly when Hudson runs behind a reverse proxy.
         String url = Jenkins.get().getRootUrl();
         try {
             if (url != null) {
-                String host = new URL(url).getHost();
+                String host = new URI(url).toURL().getHost();
                 if (host != null)
                     return host;
             }
         } catch (MalformedURLException e) {
             // fall back to HTTP request
+            throw new MalformedURLException();
```

Comment:
this is not falling back.

Reply:
removed `throw new MalformedURLException();` to enable fallback as in original code